src/schema/schema_drop.c

```diff
@@ -462,11 +476,13 @@ __schema_drop(WT_SESSION_IMPL *session, const char *uri, const char *cfg[], bool
     if (ret == WT_NOTFOUND || ret == ENOENT)
         ret = force ? 0 : ENOENT;
 
-    if (F_ISSET(S2C(session), WT_CONN_BACKUP_PARTIAL_RESTORE))
-        WT_TRET(__wt_meta_track_off(session, false, ret != 0));
-    else
-        WT_TRET(__wt_meta_track_off(session, true, ret != 0));
-
+    /*
+     * FIXME-WT-16215: During recovery (including partial backup restore), the meta tracking has not
+     * been initialized yet. We don't need to use meta tracking as recovery must end with a
+     * checkpoint to syncs all files.
+     */
+    bool need_sync = !F_ISSET(S2C(session), WT_CONN_BACKUP_PARTIAL_RESTORE | WT_CONN_RECOVERING);
+    WT_TRET(__wt_meta_track_off(session, need_sync, ret != 0));
```

Comment:
Can we simply call __wt_meta_track_on/off for __schema_drop only if `WT_CONN_RECOVERING` is not set?

Reply:
I thought this would be a two-line change that wouldn’t significantly increase the PR size. I’m just concerned that doing this with need_sync == false could be more dangerous than not doing it at all. Given the current team load, I’m also unsure when a separate ticket for this could be completed in the near term.

However, if you’re confident that the current approach is safe, I’m fine with investigating it in a separate PR.

Reply:
I'm confident that the current approach is safe. The WT_CONN_BACKUP_PARTIAL_RESTORE feature is performed in the recovery stage and has baked in MongoDB at least 1 year+. There hasn't been any issues so far